src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionSubscriber.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DBALException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use ProxyManager\Proxy\LazyLoadingInterface;
+use Symfony\Bridge\Doctrine\Event\ForceKernelRebootEvent;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\VarExporter\LazyObjectInterface;
+
+/**
+ * Based on https://github.com/Baldinof/roadrunner-bundle/blob/3.x/src/Integration/Doctrine/DoctrineORMMiddleware.php.
+ */
+class DoctrineConnectionSubscriber implements EventSubscriberInterface
+{
+    public function __construct(private ManagerRegistry $managerRegistry, private ContainerInterface $container, private EventDispatcherInterface $eventDispatcher)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event)
+    {
+        $connectionServices = $this->managerRegistry->getConnectionNames();
+
+        foreach ($connectionServices as $connectionServiceName) {
+            if (!$this->container->initialized($connectionServiceName)) {
+                continue;
+            }
+
+            $connection = $this->container->get($connectionServiceName);
+
+            if (!$connection instanceof Connection) {
+                throw new \RuntimeException(sprintf('The value "%s" is not an instance of "%s".', $connection ? $connection::class : 'null', Connection::class));
+            }
+
+            if ($connection->isConnected() && !$this->ping($connection)) {
+                $connection->close();
+            }
+
+            $managerNames = $this->managerRegistry->getManagerNames();
+
+            foreach ($managerNames as $managerName) {
+                if (!$this->container->initialized($managerName)) {
+                    continue;
+                }
+
+                $manager = $this->container->get($managerName);
+
+                if (!$manager instanceof EntityManagerInterface) {
+                    throw new \RuntimeException(sprintf('The value "%s" is not an instance of "%s".', $manager ? $manager::class : 'null', EntityManagerInterface::class));
```

Comment:
Again, you should not make it illegal to use different instances of OM. For example, this will break everything for ODM users.